modules/dreamview/backend/teleop/teleop.cc

```diff
@@ -104,23 +132,84 @@ void TeleopService::RegisterMessageHandlers() {
       });
 }
 
-void TeleopService::UpdateModemInfo(
-    const std::shared_ptr<ModemInfo> &modem_info) {
-  // TODO simplify data and only send necessary info for display
-  if (modem_info->has_provider() && modem_info->has_technology()) {
-    boost::unique_lock<boost::shared_mutex> writer_lock(mutex_);
-    teleop_status_["modems"][modem_info->provider()] = modem_info->technology();
-  }
-}
 
 void TeleopService::SendStatus(WebSocketHandler::Connection *conn) {
   std::string to_send;
   {
     boost::shared_lock<boost::shared_mutex> reader_lock(mutex_);
     to_send = teleop_status_.dump();
+    AINFO << "status: " << to_send << std::endl;
   }
   websocket_->SendData(conn, to_send);
 }
 
+
+void TeleopService::UpdateModem(unsigned int index,
+    const std::shared_ptr<ModemInfo> &modem_info) {
+
+    // AINFO << "UpdateModem: " << index << std::endl;
+    boost::unique_lock<boost::shared_mutex> writer_lock(mutex_);
+    // TODO simplify data and only send necessary info for display
+    // update modem_info_
+    if (modem_info->has_provider() && modem_info->has_technology()) {
+        boost::unique_lock<boost::shared_mutex> writer_lock(mutex_);
+        std::string modemId = "0";
+        if (index == 0) {
+            modemId = "1";
+        }
+        if (index == 1) {
+            modemId = "2";
+        }
+        if (index == 2) {
+            modemId = "3";
+        }
+
+        // teleop_status_["modems"][modem_info->provider()] =
+        //  modem_info->technology();
+        teleop_status_["modems"][modemId] = modem_info->technology();
+    }
+}
+
+void TeleopService::UpdateCarDaemonRpt(
+    const std::shared_ptr<DaemonServiceRpt> &daemon_rpt) {
+  {
+      boost::unique_lock<boost::shared_mutex> writer_lock(mutex_);
+      // AINFO << "CarDaemonRpt" << std::endl;
+      bool aVideoEncoderIsRunning = false;
+      bool voipIsRunning = false;
+      for (int i = 0; i < daemon_rpt->services_size(); i++) {
+          std::string service =  daemon_rpt->services(i);
+          AINFO <<  "  *" << service << std::endl;
+          if (service.find("encoder") >= 0) {
```

Comment:
if this is checked before 'voip_encoder', it will set 'aVideoEncoderIsRunning' even if it finds 'voip_encoder'.

Reply:
this issue is not addressed yet